src/main/java/io/vertx/redis/impl/RedisConnection.java

```diff
@@ -279,6 +254,88 @@ void send(final Command<?> command) {
     }
   }
 
+  /**
+   * Once a socket connection is established one needs to authenticate if there is a password
+   */
+  private synchronized void doAuth() {
+    if (config.getAuth() != null) {
+      // we need to authenticate first
+      Command<String> authCmd = new Command<>(RedisCommand.AUTH, new JsonArray().add(config.getAuth()), Charset.forName(config.getEncoding()), ResponseTransform.NONE, String.class).handler(auth -> {
+        if (auth.failed()) {
+          Command<?> cmd;
+
+          // clean up any pending command
+          while ((cmd = pending.poll()) != null) {
+            cmd.handle(Future.failedFuture(auth.cause()));
+          }
+
+          netSocket.close();
+          state = State.DISCONNECTED;
+          return;
+        }
+
+        // auth success, proceed with select
+        doSelect();
+      });
+
+      // queue it
+      waiting.add(authCmd);
+      // write to the socket in the netSocket context
+      context.runOnContext(v -> authCmd.writeTo(netSocket));
+    } else {
+      // no auth, proceed with select
+      doSelect();
+    }
+  }
+
+  private synchronized void doSelect() {
+    // optionally there could be a select command
+    if (config.getSelect() != null) {
+      Command<String> selectCmd = new Command<>(RedisCommand.SELECT, new JsonArray().add(config.getSelect()), Charset.forName(config.getEncoding()), ResponseTransform.NONE, String.class).handler(select -> {
+        if (select.failed()) {
+          Command<?> cmd;
+
+          // clean up any pending command
+          while ((cmd = pending.poll()) != null) {
+            cmd.handle(Future.failedFuture(select.cause()));
+          }
+
+          netSocket.close();
+          state = State.DISCONNECTED;
+          return;
+        }
+
+        // select success, proceed with resend
+        resendPending();
+      });
+
+      // queue it
+      waiting.add(selectCmd);
+      // write to the socket in the netSocket context
+      context.runOnContext(v -> selectCmd.writeTo(netSocket));
+    } else {
+      // no select, proceed with resend
+      resendPending();
+    }
+  }
+
+  private synchronized void resendPending() {
+    Command<?> command;
+
+    // we are connected so clean up the pending queue
+    while ((command = pending.poll()) != null) {
+      // The order read must match the order written, vertx guarantees
+      // that this is only called from a single thread.
+      for (int i = 0; i < command.getExpectedReplies(); ++i) {
+        waiting.add(command);
+      }
+
+      // write to the socket in the netSocket context
+      final Command<?> _command = command;
+      context.runOnContext(v -> _command.writeTo(netSocket));
```

Comment:
not sure this is needed, you should be able to write to a NetSocket outside of the context.